src/components/dashboard/menu-table/dropdown.js

```diff
@@ -53,43 +65,28 @@ class Dropdown extends React.Component {
         }
     }
 
-    showDropdown (event) {
-        event.preventDefault()
-        console.log("showDropdown")
-        this.setState({displayList: true}, () => {
-            document.addEventListener('click', this.hideDropdown)
-        });
-    }
-
-    hideDropdown () {
-        this.setState({displayList: false}, () => {
-            document.removeEventListener('click', this.hideDropdown)
-        });
-    }
-
-    select (item) {
-        this.setState({currentSelection: item.name});
-        this.props.updateSelection(item)
+    getCategory = (event, { value }) => {
+        let categoryId = this.state.categoryNames.indexOf(value) + 1;
+        this.props.updateSelection(categoryId);
+        this.state.currCategory = this.state.categoryNames[categoryId - 1];
```

Comment:
I don't think this is the correct way to do things because category ids might start from any number. I think ur assuming category ids are [1, 2...]? Maybe you want another array list categoryNames or just an array of Category JSON objects.